examples/reaction-roles/src/index.ts

```diff
@@ -0,0 +1,47 @@
+import { createBot } from '@discordeno/bot'
+import { config } from 'dotenv'
+
+import commands from './commands/index.js'
+import events from './events/index.js'
+
+config()
+const token = process.env.TOKEN
+
+// Ensure the existence of the TOKEN env
+if (!token) throw new Error('The TOKEN environment variable needs to be defined.')
+
+export const bot = createBot({
+  token,
+  events,
+})
+
+// Setup for the desiredProperties
+
+bot.transformers.desiredProperties.user.id = true
+
+bot.transformers.desiredProperties.message.id = true
+
+bot.transformers.desiredProperties.member.roles = true
+
+bot.transformers.desiredProperties.interaction.id = true
+bot.transformers.desiredProperties.interaction.data = true
+bot.transformers.desiredProperties.interaction.type = true
+bot.transformers.desiredProperties.interaction.user = true
+bot.transformers.desiredProperties.interaction.token = true
+bot.transformers.desiredProperties.interaction.member = true
+bot.transformers.desiredProperties.interaction.message = true
+bot.transformers.desiredProperties.interaction.guildId = true
+bot.transformers.desiredProperties.interaction.channelId = true
+
+bot.transformers.desiredProperties.role.id = true
+
+const guildId = 'REPLACE WITH YOUR GUILD ID'
+
+// `upsertGuildApplicationCommands` is a promise, but if we await and we got ratelimited from the discord endpoint it can force the code to wait for the ratelimit end
+bot.rest
+  .upsertGuildApplicationCommands(guildId, [...commands.values()])
+  .catch((e) => bot.logger.error('There was an error when updating the global commands', e))
```

Comment:
Don't think update every time is good idea, might hit rate limit

Reply:
For me personally, I upsert each time but I also know how many times I reboot and the bot is private. 
However, if we're using this as a template, maybe having something that stores the guild ID and if the command has been registered? I'm pretty sure that's the standard practice no? In some way or form.

Reply:
I have a bot that is public and i upsert each time, in dev i do get ratelimited a bit but it is not that big of a deal for me in that bit

Reply:
exact same situation here.

I don't know what exactly the right answer is. I see a couple potential paths forward.

1. Don't change it and we just document it with a code comment.
2. Add some simple system (SQLite or like a JSON file DB style thing) that just keeps track in case someone uses the template for a bigger bot.

Reply:
I have decided to add a comment and a note under the `Improvements` section of the tutorial because:
- I don't want to complicate the code to run if you want to run the example on your own to test it out and play with the code
- I don't want to complicate the code to check for a JSON file (or worse provb on a code prospective, SQLite) knowing the previous state of the commands and conditionally upsert the commands

Let me know if you think the comment is not clear enough (or the note in the .md file)